core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -467,42 +471,79 @@ private void processProviderRequest(TransferProcess process, DataRequest dataReq
         }
     }
 
+    private boolean processConsumerRequest(TransferProcess process, DataRequest dataRequest) {
+        int retryCount = process.getStateCount() - 1;
```

Comment:
just want to understand, why we need to subtract 1 here ?

Reply:
If the state count is 2, it means it's been retried once